app/routers/books.py

```diff
@@ -57,3 +61,28 @@ async def create_book(book: BookCreateModel, session=Depends(get_session)):
 
     return BookModel.from_orm(book)
 
+
+@router.put("/books/{book_id}", response_model=BookModel, status_code=200)
+async def update_book(book_id: int, book_update: BookUpdate, session=Depends(get_session)):
+    for var, value in vars(book_update).items():
```

Comment:
I don't see any query, when you get the book from database.
if book is found, you should update it, otherwise return 404.